statsmodels/discrete/discrete_model.py

```diff
@@ -747,21 +747,20 @@ def predict(self, params, exog=None, exposure=None, offset=None,
         #TODO: add offset tp
         if exog is None:
             exog = self.exog
-            offset = getattr(self, 'offset', 0)
+        
+        if exposure is None:
             exposure = getattr(self, 'exposure', 0)
-
         else:
-            if exposure is None:
-                exposure = 0
-            else:
-                exposure = np.log(exposure)
-            if offset is None:
-                offset = 0
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = getattr(self, 'offset', 0)
 
+        Xb = np.dot(exog, params[:exog.shape[1]])
```

Comment:
Just pushed commits to implement all your suggestions except for this one.  I like Xb because it is a stand-in for X\beta which reminds me of my youth.